hack/generator/pkg/astmodel/interface.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+// Interface defines an interface implementation
+type Interface struct {
```

Comment:
Probably. Will fix.

Reply:
Okay so I think we shouldn't call this `InterfaceType` because that suggests that this is about defining an `interface` (`ObjectType` is about defining an object/representing an object).

I've changed this to `InterfaceImplementation` because this is about an implementation of an `interface` (which in Go is basically a collection of functions). I think it resolves the issue and also side-steps the "is this about creating an interface or about implementing it" issue.

Reply:
Oh yeah, that's much nicer - I think that explains some confusion I had looking at the handling down in ObjectType as well.